### DISCUSSION TITLE
Overwriting Request Body for Contract Tests

### DISCUSSION THREAD

**Discussion by PaulCluver** [apideck-libraries/portman#503] [Q&A] [answered]
Hi there, 

I have a use case wherein I need to overwrite a default value that is stored in my open api spec. In my api spec, I set a default value for identificationNumber. I have certain business rules in place which preclude the repeated usage of this number. Basically, our system allows for the creation of only 1 application for a given ID number. While this is very simple to do in the context of a variation test, I have not found a way to do this in a contract test. 

Given that overwriteRequestBody does not appear to be accessible within a contract test, does there exist a technique to affect such an override at the contractTest level? 

Kind regards,
Paul.

**Comment by thim81** [comment]
Not 100% sure of your use-case, but would setting the ID number as a variable be a solution?

That way you could define a variable [upfront](https://github.com/apideck-libraries/portman#environment-variables-as-postman-variables) or [dynamically](https://github.com/apideck-libraries/portman#portman---assignvariables-properties) and use that in the [request overwrite](https://github.com/apideck-libraries/portman#portman---overwrites-properties).

**Reply by PaulCluver** [reply]
Hi Tim, 

Setting the ID number as a variable is an option that I previously explored. However, it ain't working. I can only assume that there is something incorrect in my configuration. Perhaps you can spot the issue here: 

**Contract Test Config.** 
![image](https://github.com/apideck-libraries/portman/assets/5680263/0c20b89d-929e-4e9d-8ab8-f2c2c030023c)

**Exception**
![image](https://github.com/apideck-libraries/portman/assets/5680263/e99ada6c-a871-4596-a605-a5f1739547c9)

Kind regards,
Paul

**Reply by thim81** [reply, marked as answer]
hi @PaulCluver 

The **overwrites** have a separate section in the config. Since the overwrites will be applied for all "tests".
Have a look at an example:

https://github.com/apideck-libraries/portman/blob/main/portman-config.example.json

the structure is:
- tests
  - contractTests
  - contentTests
- assignVariables
- overwrites
- globals

**Comment by PaulCluver** [comment]
The above suggestion works perfectly. Thanks for the support
